packages/runtime/src/widgets/Date.tsx

```diff
@@ -0,0 +1,361 @@
+            open={openPicker}
+            maxDate={props.lastDate ? dayjs(props.lastDate) : undefined}
+            minDate={props.firstDate ? dayjs(props.firstDate) : undefined}
+            onClose={(): void => setOpenPicker(false)}
+            onAccept={onDateAccept}
+            onChange={onChangeCallback}
+            slots={{
+              calendarHeader: CustomCalendarHeader,
+              actionBar: CustomActionBar,
+            }}
+            slotProps={{
+              calendarHeader: {
+                labelId: value?.format(DateHeaderFormat),
+              },
+              actionBar: {
+                actions: ["cancel", "accept"],
+              },
+              field: {
+                clearable: Boolean(value),
+                onClear: () => setValue(undefined),
+              },
+              textField: {
+                InputLabelProps: {
+                  shrink: false,
+                },
+                onClick: () => setOpenPicker(true),
+                error: false,
+                color: "success",
+              },
+              popper: {
+                modifiers: [
+                  {
+                    name: "flip",
+                    enabled: false,
+                  },
+                ],
+              },
+              layout: {
+                sx: {
+                  "& .MuiDateCalendar-root": {
+                    overflow: "visible !important",
+                    height: "unset !important",
+                    maxHeight: "unset !important",
+                  },
+                  "& .MuiPickersFadeTransitionGroup-root.MuiDateCalendar-viewTransitionContainer":
+                    {
+                      display: `${
+                        isCalendarOpen ? "block" : "none"
+                      } !important`,
+                    },
+                },
+              },
+            }}
+            sx={{
+              "& .MuiOutlinedInput-notchedOutline": {
+                top: "0px",
+              },
+              "& .MuiOutlinedInput-notchedOutline legend": {
+                display: "none",
+              },
+              "& .MuiInputBase-input.MuiOutlinedInput-input.MuiInputBase-inputAdornedEnd":
+                {
+                  visibility: `${!value ? "hidden" : "visible"} !important`,
+                },
+              "& .MuiDateCalendar-root": {
+                overflow: "visible !important",
+                height: "unset !important",
+                maxHeight: "unset !important",
+              },
+              "& .MuiPickersFadeTransitionGroup-root.MuiDateCalendar-viewTransitionContainer":
+                {
+                  display: "none !important",
+                },
+            }}
+          />
+        </LocalizationProvider>
+      </AntForm.Item>
+    </DatePickerContext.Provider>
+  );
+};
+
+WidgetRegistry.register("Date", Date);
+
+const CustomCalendarHeader: React.FC<PickersCalendarHeaderProps<any>> = (
```

Comment:
Why is this extra component needed?

Reply:
these extra components are being used here https://github.com/EnsembleUI/ensemble-react/blob/d6c200c43766f6c2db8c3064d8863df8e0fbd3f4/packages/runtime/src/widgets/Date.tsx#L130-L133